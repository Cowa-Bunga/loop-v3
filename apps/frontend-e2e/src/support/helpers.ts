export const testIdSelector = (testId: string) => `[data-testid="${testId}"]`

export const COOKIE_NAME = 'cowabunga-user-cookie'
