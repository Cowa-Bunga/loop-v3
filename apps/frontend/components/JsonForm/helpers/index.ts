const makeInputId = (id: string, suffix?: string) => {
  const parts = id.split('/')

  if (suffix) {
    return `${parts[parts.length - 1]}-${suffix}`
  }

  return parts[parts.length - 1]
}

export { makeInputId }
