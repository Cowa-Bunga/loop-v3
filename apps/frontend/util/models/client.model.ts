import defaultsDeep from 'lodash/defaultsDeep'

export const modelClientList = (
  data: { [s: string]: unknown } | ArrayLike<unknown>
): IClient[] => {
  return Object.entries(data)
    .map(modelClientData)
    .map((el: never) => el['1'])
    .sort((a: IClient, b: IClient) => a.name.localeCompare(b.name))
}

export const modelClientData = (data): IClient => {
  return defaultsDeep(data, emptyClientModel)
}
export const emptyClientModel: IClient = {
  client_id: '',
  name: '',
  user_id: '',
  permissions_id: '',
  logo: '',
  permissions: {
    scopes: [],
    fleet: false,
    administrator: false
  }
}
