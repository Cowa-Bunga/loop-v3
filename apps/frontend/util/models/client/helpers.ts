import { defaultsDeep } from 'lodash'
import { MODEL_CLIENT_EMPTY } from './model'

export const modelClientList = (
  data: { [s: string]: unknown } | ArrayLike<unknown>
): IClient[] => {
  return Object.entries(data)
    .map(modelClientData)
    .map((el: never) => el['1'])
    .sort((a: IClient, b: IClient) => a.name.localeCompare(b.name))
}

export const modelClientData = (data: unknown): IClient => {
  return defaultsDeep(data, MODEL_CLIENT_EMPTY)
}
