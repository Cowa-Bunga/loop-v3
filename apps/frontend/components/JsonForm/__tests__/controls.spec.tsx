/* eslint-disable @typescript-eslint/ban-ts-comment */
import { renderWithProviders } from '@test/helpers'
import {
  AutoCompleteControl,
  CheckboxControl,
  DateTimeControl,
  DropdownControl,
  GooglePlacesControl,
  MultiSelectControl,
  PhoneNumberControl,
  TextInputControl
} from '../controls'
import { UISchemaElement } from '@jsonforms/core/src/models/uischema'
import { fireEvent } from '@testing-library/react'

const mockControlData = {
  data: [],
  handleChange: jest.fn(),
  path: 'test',
  id: 'test-id',
  schema: {
    enum: ['test1', 'test2'],
    options: { api_key: '' }
  },
  label: 'test-label',
  required: true,
  visible: true,
  uischema: {
    type: 'Control',
    scope: '#/properties/test',
    options: {},
    label: 'test-label',
    rule: {
      effect: 'SHOW',
      condition: {
        type: 'LEAF'
      }
    }
  } as UISchemaElement
}

jest.mock('@jsonforms/react', () => ({
  withJsonFormsControlProps: jest
    .fn()
    .mockImplementation((Component) => Component)
}))

describe('JsonForm Controls', () => {
  describe('AutoCompleteControl', () => {
    it('should render', () => {
      const { baseElement } = renderWithProviders(
        // @ts-ignore
        <AutoCompleteControl {...mockControlData} />
      )

      expect(baseElement).toMatchSnapshot()
    })
  })

  describe('CheckboxControl', () => {
    it('should render', () => {
      const { baseElement } = renderWithProviders(
        // @ts-ignore
        <CheckboxControl {...mockControlData} />
      )

      expect(baseElement).toMatchSnapshot()
    })

    it('should trigger handleChange', () => {
      const { baseElement } = renderWithProviders(
        // @ts-ignore
        <CheckboxControl {...mockControlData} />
      )

      const checkbox = baseElement.querySelector('input[type="checkbox"]')
      fireEvent.click(checkbox)
      expect(mockControlData.handleChange).toHaveBeenCalled()
    })
  })

  describe('DateTimeControl', () => {
    it('should render', () => {
      const { baseElement } = renderWithProviders(
        // @ts-ignore
        <DateTimeControl {...mockControlData} />
      )

      expect(baseElement).toMatchSnapshot()
    })
  })

  describe('DropdownControl', () => {
    it('should render', () => {
      const { baseElement } = renderWithProviders(
        // @ts-ignore
        <DropdownControl {...mockControlData} />
      )

      expect(baseElement).toMatchSnapshot()
    })
  })

  xdescribe('GooglePlacesControl', () => {
    it('should render', () => {
      const { baseElement } = renderWithProviders(
        // @ts-ignore
        <GooglePlacesControl {...mockControlData} />
      )

      expect(baseElement).toMatchSnapshot()
    })
  })

  describe('MultiSelectControl', () => {
    it('should render', () => {
      const { baseElement } = renderWithProviders(
        // @ts-ignore
        <MultiSelectControl {...mockControlData} />
      )

      expect(baseElement).toMatchSnapshot()
    })
  })

  describe('PhoneNumberControl', () => {
    it('should render', () => {
      const { baseElement } = renderWithProviders(
        // @ts-ignore
        <PhoneNumberControl {...mockControlData} />
      )

      expect(baseElement).toMatchSnapshot()
    })
  })

  describe('TextInputControl', () => {
    it('should render', () => {
      const { baseElement } = renderWithProviders(
        // @ts-ignore
        <TextInputControl {...mockControlData} />
      )

      expect(baseElement).toMatchSnapshot()
    })
  })
})
