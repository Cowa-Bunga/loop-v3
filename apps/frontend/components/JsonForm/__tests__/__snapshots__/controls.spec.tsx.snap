// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JsonForm Controls AutoCompleteControl should render 1`] = `
<body>
  <div>
    <div
      id="test-id"
    >
      <div
        class="MuiFormControl-root MuiFormControl-fullWidth css-cq5a0p-MuiFormControl-root"
      >
        <div
          class="MuiAutocomplete-root MuiAutocomplete-hasPopupIcon css-1h51icj-MuiAutocomplete-root"
        >
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              for="test-id-auto-complete"
              id="test-id-auto-complete-label"
            >
              test-label
              <span
                aria-hidden="true"
                class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
              >
                 
                *
              </span>
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiAutocomplete-inputRoot css-154xyx0-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-autocomplete="list"
                aria-expanded="false"
                aria-invalid="false"
                autocapitalize="none"
                autocomplete="off"
                class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
                id="test-id-auto-complete"
                required=""
                role="combobox"
                spellcheck="false"
                type="text"
                value=""
              />
              <div
                class="MuiAutocomplete-endAdornment css-1q60rmi-MuiAutocomplete-endAdornment"
              >
                <button
                  aria-label="Open"
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-popupIndicator css-qzbt6i-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-popupIndicator"
                  tabindex="-1"
                  title="Open"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="ArrowDropDownIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M7 10l5 5 5-5z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-yjsfm1"
                >
                  <span>
                    test-label
                     
                    *
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`JsonForm Controls CheckboxControl should render 1`] = `
<body>
  <div>
    <div
      id="test-id"
    >
      <div
        class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
        style="margin: 23px 0px;"
      >
        <label
          class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
          id="test-id-checkbox"
        >
          <span
            class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary Mui-checked MuiCheckbox-root MuiCheckbox-colorPrimary css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
          >
            <input
              checked=""
              class="PrivateSwitchBase-input css-1m9pwf3"
              data-indeterminate="false"
              type="checkbox"
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="CheckBoxIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </span>
          <span
            class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
          >
            test-label
          </span>
        </label>
      </div>
    </div>
  </div>
</body>
`;

exports[`JsonForm Controls DateTimeControl should render 1`] = `
<body>
  <div>
    <div
      id="test-id"
    >
      <div
        class="MuiFormControl-root MuiFormControl-fullWidth css-cq5a0p-MuiFormControl-root"
      >
        <div
          class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-error Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            for=":r2:"
            id=":r2:-label"
          >
            test-label
            <span
              aria-hidden="true"
              class="MuiFormLabel-asterisk MuiInputLabel-asterisk Mui-error css-wgai2y-MuiFormLabel-asterisk"
            >
               
              *
            </span>
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary Mui-error MuiInputBase-formControl MuiInputBase-adornedEnd css-o9k5xi-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="true"
              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
              id=":r2:"
              placeholder="mm/dd/yyyy hh:mm (a|p)m"
              required=""
              type="tel"
              value=""
            />
            <div
              class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
            >
              <button
                aria-label="Choose date"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="CalendarIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-yjsfm1"
              >
                <span>
                  test-label
                   
                  *
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`JsonForm Controls DropdownControl should render 1`] = `
<body>
  <div>
    <div
      id="test-id"
    >
      <div
        class="MuiFormControl-root MuiFormControl-fullWidth css-cq5a0p-MuiFormControl-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          id="test-id-label"
        >
          test-label
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl  css-1yk1gt9-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
          required=""
        >
          <div
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-labelledby="test-id-label test-id-select"
            class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
            id="test-id-select"
            role="button"
            tabindex="0"
          >
            <span
              class="notranslate"
            >
              ​
            </span>
          </div>
          <input
            aria-hidden="true"
            aria-invalid="false"
            class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
            required=""
            tabindex="-1"
            value=""
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
            data-testid="ArrowDropDownIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-yjsfm1"
            >
              <span>
                test-label
                 
                *
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`JsonForm Controls GooglePlacesControl should render 1`] = `
<body>
  <div>
    <div
      id="test-id"
    >
      <div
        class="MuiFormControl-root MuiFormControl-fullWidth css-cq5a0p-MuiFormControl-root"
      >
        <div />
      </div>
    </div>
  </div>
</body>
`;

exports[`JsonForm Controls MultiSelectControl should render 1`] = `
<body>
  <div>
    <div
      id="test-id"
    >
      <div
        class="MuiFormControl-root MuiFormControl-fullWidth css-cq5a0p-MuiFormControl-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          id="test-id-label"
        >
          test-label
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl  css-1yk1gt9-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
          required=""
        >
          <div
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-labelledby="test-id-label test-id-select"
            class="MuiSelect-select MuiSelect-outlined MuiSelect-multiple MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
            id="test-id-select"
            role="button"
            tabindex="0"
          >
            <span
              class="notranslate"
            >
              ​
            </span>
          </div>
          <input
            aria-hidden="true"
            aria-invalid="false"
            class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
            required=""
            tabindex="-1"
            value=""
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
            data-testid="ArrowDropDownIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-yjsfm1"
            >
              <span>
                test-label
                 
                *
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`JsonForm Controls PhoneNumberControl should render 1`] = `
<body>
  <div>
    <div
      id="test-id"
    >
      <div
        class="MuiFormControl-root MuiFormControl-fullWidth css-cq5a0p-MuiFormControl-root"
      >
        <div
          class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined css-1jy569b-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="true"
            for="test-id-text-input"
            id="test-id-text-input-label"
          >
            test-label
            <span
              aria-hidden="true"
              class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
            >
               
              *
            </span>
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedStart css-1q6at85-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <div
              class="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-ittuaa-MuiInputAdornment-root"
              style="position: relative;"
            >
              <span
                class="notranslate"
              >
                ​
              </span>
              <button
                aria-haspopup="true"
                aria-label="Select country"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                style="min-width: 30px; padding: 0px; height: 30px;"
                tabindex="0"
                type="button"
              >
                <svg
                  class="margin"
                  viewBox="0 85.333 512 341.333"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0 85.337h512v341.326H0z"
                    fill="#FFF"
                  />
                  <path
                    d="M114.024 256.001 0 141.926v228.17z"
                  />
                  <path
                    d="M161.192 256 0 94.7v47.226l114.024 114.075L0 370.096v47.138z"
                    fill="#ffb915"
                  />
                  <path
                    d="M509.833 289.391c.058-.44.804-.878 2.167-1.318v-65.464H222.602L85.33 85.337H0V94.7L161.192 256 0 417.234v9.429h85.33l137.272-137.272h287.231z"
                    fill="#007847"
                  />
                  <path
                    d="M503.181 322.783H236.433l-103.881 103.88H512v-103.88z"
                    fill="#000c8a"
                  />
                  <path
                    d="M503.181 189.217H512V85.337H132.552l103.881 103.88z"
                    fill="#e1392d"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedStart css-1o9s3wi-MuiInputBase-input-MuiOutlinedInput-input"
              id="test-id-text-input"
              placeholder="+1 (702) 123-4567"
              required=""
              type="tel"
              value="+27"
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-14lo706"
              >
                <span>
                  test-label
                   
                  *
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`JsonForm Controls TextInputControl should render 1`] = `
<body>
  <div>
    <div
      id="test-id"
    >
      <div
        class="MuiFormControl-root MuiFormControl-fullWidth css-cq5a0p-MuiFormControl-root"
      >
        <div
          class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            for="test-id-text-input"
            id="test-id-text-input-label"
          >
            test-label
            <span
              aria-hidden="true"
              class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
            >
               
              *
            </span>
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              id="test-id-text-input"
              required=""
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-yjsfm1"
              >
                <span>
                  test-label
                   
                  *
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
